pjsip/src/pjsua-lib/pjsua_media.c

```diff
@@ -1624,14 +1638,29 @@ pj_status_t call_media_on_event(pjmedia_event *event,
     }
 
     if (pjsua_var.ua_cfg.cb.on_call_media_event) {
-	if (call) {
-	    (*pjsua_var.ua_cfg.cb.on_call_media_event)(call->index,
-						       call_med->idx, event);
-	} else {
+	pjsua_event_list *eve = NULL;
+ 
+    	pj_mutex_lock(pjsua_var.timer_mutex);
+
+    	if (pj_list_empty(&pjsua_var.event_list)) {
+            eve = PJ_POOL_ALLOC_T(pjsua_var.pool, pjsua_event_list);
```

Comment:
after a quick check, access to `pjsua_var.pool` should be protected by PJSUA lock, but unfortunately locking PJSUA lock is the cause of the deadlock, so the lock must be avoided. A solution is to use a specific pool which can be shared with pjsua timer (note that the pjsua timer also uses `pjsua_var.pool`, so it is buggy, but it can simply use PJSUA lock without a risk of deadlock).